File: app/src/main/java/com/jbrunton/mymovies/api/resources/ConfigurationResponse.java
package com.jbrunton.mymovies.api.resources;

import com.jbrunton.entities.Configuration;

public class ConfigurationResponse {
    private ConfigurationImageResource images;

    public Configuration toModel() {
        return Configuration.create(images.secureBaseUrl);
    }

    static class ConfigurationImageResource {
        private String secureBaseUrl;
    }
}

File: networking/src/main/java/com/jbrunton/mymovies/networking/resources/configuration/ConfigurationResponse.kt
package com.jbrunton.mymovies.networking.resources.configuration

import com.jbrunton.mymovies.entities.models.Configuration

data class ConfigurationResponse(private val images: ConfigurationImageResource) {

    fun toModel() = Configuration(images.secureBaseUrl)

    data class ConfigurationImageResource(val secureBaseUrl: String)
}
